File: app/src/main/java/com/github/hurshi/AppComponent.java
package com.github.hurshi;

import javax.inject.Singleton;

import dagger.Component;
import github.hurshi.daggervminjector.extension.AndroidDaggerVMInjectionModule;

/**
 * Created by gavin on 2017/10/24.
 */

@Singleton
@Component(modules = {AndroidDaggerVMInjectionModule.class
        , _GPDaggerVMInjectModule.class
})
public interface AppComponent {
    void inject(App app);
}

File: app_kotlin/src/main/java/com/github/hurshi/app_kotlin/AppComponent.kt
package com.github.hurshi.app_kotlin

import dagger.Component
import github.hurshi.daggervminjector.extension.AndroidDaggerVMInjectionModule
import javax.inject.Singleton

/**
 * Created by gavin on 2017/10/24.
 */

@Singleton
@Component(modules = arrayOf(AndroidDaggerVMInjectionModule::class
        , _GPDaggerVMInjectModule::class
))
interface AppComponent {
    fun inject(app: App)
}
